pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentDispatcherMultipleConsumers.java

```diff
@@ -235,7 +235,7 @@ public synchronized void removeConsumer(Consumer consumer) throws BrokerServiceE
 
     @Override
     public void consumerFlow(Consumer consumer, int additionalNumberOfMessages) {
-        topic.getBrokerService().executor().execute(() -> {
+        topic.getOrderedExecutor().execute(() -> {
```

Comment:
Does it seem to be possible to use `this.dispatchMessagesThread`?

Reply:
I don't know why `this.dispatchMessagesThread ` is not the current topic thread.